File: text_readability_grader/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    char* letras = get_string("Digite o texto a ser avaliado: \n");

    int qtd_letras = 0;
    int qtd_palavras = 1;
    int qtd_sentencas = 0;
    int index = 0;
    while (index < strlen(letras))
    {
        if (( (char) letras[index] >= 65 && (char) letras[index] <= 90) || ((char) letras[index] >= 97 && (char) letras[index] <= 122))
        {
            qtd_letras ++;
        }
        if ((char) letras[index] == ' ')
        {
            qtd_palavras ++;
        }
        if ((char) letras[index] == '.' || (char) letras[index] == '?' || (char) letras[index] == '!')
        {
            qtd_sentencas ++;
        }
        index ++;
    }

    float l = qtd_letras * 100 / (float) qtd_palavras;
    float s = qtd_sentencas * 100 / (float) qtd_palavras;
    float indice = 0.0588 * l - 0.296 * s - 15.8;
    
    if ((int) round(indice) >= 16)
    {
        printf("Grade 16+\n");
    }
    else if ((int) round(indice) < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(indice));
    }
}

File: text_readability_grader/readability.py
import re

while True:
    try:
        texto = input("Digite o texto a ser avaliado: ")
        if texto == False:
            raise TypeError
        break
    except TypeError:
        print("Nenhum texto digitado")


def n_palavras(texto):
    palavras = texto.split(" ")
    return len(palavras)


def n_sentencas(texto):
    sentencas = re.split("[.?!]", texto)
    return len(sentencas) - 1


def n_letras(texto):
    caracteres = tuple(texto)
    letras = []
    for caractere in caracteres:
        if caractere == caractere.upper() and caractere == caractere.lower():
            pass
        else:
            letras.append(caractere)
    return len(letras)


l = n_letras(texto) * 100 / n_palavras(texto)
s = n_sentencas(texto) * 100 / n_palavras(texto)
indice = 0.0588 * l - 0.296 * s - 15.8
if indice >= 16:
    print("Grade 16+\n")
elif indice < 1:
    print("Before Grade 1\n")
else:
    print(f"Grade {round(indice)}")
